File: android/app/src/main/java/com/kreativekoala/cleanup/data/model/ContactModels.kt
package com.kreativekoala.cleanup.data.model

import java.util.UUID

data class ContactItem(
    val contactId: Long,
    val rawContactId: Long = 0,
    val displayName: String,
    val phoneNumbers: List<String> = emptyList(),
    val emailAddresses: List<String> = emptyList(),
    val photoUri: String? = null
) {
    val isIncomplete: Boolean
        get() = displayName.isBlank() || (phoneNumbers.isEmpty() && emailAddresses.isEmpty())

    val initials: String
        get() = displayName.take(1).uppercase().ifEmpty { "?" }
}

data class DuplicateContactGroup(
    val id: UUID = UUID.randomUUID(),
    val contacts: List<ContactItem>,
    val matchReason: MatchReason
) {
    val duplicateCount: Int get() = contacts.size - 1
    val primaryContact: ContactItem get() = contacts.first()
}

enum class MatchReason(val label: String) {
    SAME_NAME("Same Name"),
    SAME_PHONE("Same Phone"),
    SAME_EMAIL("Same Email")
}

data class ContactScanResult(
    val count: Int,
    val groups: List<DuplicateContactGroup>
)

data class IncompleteContactResult(
    val count: Int,
    val contacts: List<ContactItem>
)

File: ios/Cleanup/Models/ContactModels.swift
import Foundation
import Contacts

struct ContactItem: Identifiable, Hashable {
    let id: String
    let contact: CNContact
    var isSelected: Bool = false

    var fullName: String {
        CNContactFormatter.string(from: contact, style: .fullName) ?? "No Name"
    }

    var phoneNumbers: [String] {
        contact.phoneNumbers.map { $0.value.stringValue }
    }

    var emailAddresses: [String] {
        contact.emailAddresses.map { $0.value as String }
    }

    var hasImage: Bool {
        contact.imageDataAvailable
    }

    var isIncomplete: Bool {
        fullName.isEmpty || (phoneNumbers.isEmpty && emailAddresses.isEmpty)
    }

    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }

    static func == (lhs: ContactItem, rhs: ContactItem) -> Bool {
        lhs.id == rhs.id
    }
}

struct DuplicateContactGroup: Identifiable {
    let id = UUID()
    var contacts: [ContactItem]
    var mergedContact: CNMutableContact?

    var matchReason: MatchReason

    enum MatchReason: String {
        case sameName = "Same Name"
        case samePhone = "Same Phone"
        case sameEmail = "Same Email"
        case similar = "Similar"
    }
}

struct ContactScanResult {
    let count: Int
    let groups: [DuplicateContactGroup]
}

struct IncompleteContactResult {
    let count: Int
    let contacts: [ContactItem]
}
